src/common.browser/RestMessageAdapter.ts

```diff
@@ -0,0 +1,279 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import {
+    ArgumentNullError,
+    ConnectionErrorEvent,
+    ConnectionEvent,
+    ConnectionMessage,
+    ConnectionMessageReceivedEvent,
+    ConnectionMessageSentEvent,
+    ConnectionOpenResponse,
+    createNoDashGuid,
+    Deferred,
+    Events,
+    EventSource,
+    MessageType,
+    Promise,
+    PromiseHelper,
+    Queue,
+} from "../common/Exports";
+import { IRequestOptions } from "./Exports";
+
+// Node.JS specific xmlhttprequest / browser support.
+import * as XHR from "xmlhttprequest-ts";
+
+export enum RestRequestType {
+    Get = "get",
+    Post = "post",
+    Delete = "delete",
+    File = "file",
+}
+
+export interface IRestResponse {
+    ok: boolean;
+    status: number;
+    statusText: string;
+    data: string;
+    json: <T>() => T;
+    headers: string;
+}
+
+interface ISendItem {
+    Message: ConnectionMessage;
+    RawMessage: XMLHttpRequest | XHR.XMLHttpRequest;
+    sendStatusDeferral: Deferred<boolean>;
+}
+
+// accept rest operations via request method and return abstracted objects from server response
+export class RestMessageAdapter {
+
+    private privTimeout: number;
+    private privIgnoreCache: boolean;
+    private privHeaders: { [key: string]: string; };
+    private privConnectionEvents: EventSource<ConnectionEvent>;
+    private privConnectionId: string;
+    private privReceivedQueue: Queue<ConnectionMessage>;
+    private privSendQueue: Queue<ISendItem>;
+
+    public constructor(
+        configParams: IRequestOptions,
+        connectionId?: string
+        ) {
+
+        if (!configParams) {
+            throw new ArgumentNullError("configParams");
+        }
```

Comment:
Look for Contracts.ThrowIfNull

Reply:
Would rather not add a dependency on sdk module if it can be avoided.